src/rulesets/Base/mapreduce.jl

```diff
@@ -10,9 +10,10 @@ function rrule(::typeof(sum), x::AbstractArray{T}; dims=:) where {T<:Number}
     y = sum(x; dims=dims)
     function sum_pullback(ȳ)
         # broadcasting the two works out the size no-matter `dims`
-        x̄ = broadcast(x, ȳ) do xi, ȳi
-            ȳi
-        end
+        x̄ = InplaceableThunk(
+            @thunk(broadcast(last∘tuple, x, ȳ)),
+            x -> x .+= ȳ
```

Comment:
This reminded me we to do https://github.com/JuliaDiff/ChainRulesCore.jl/pull/348
but i think even once that is merged this is still better (also that will not always work if dims are different, which maybe should change https://github.com/JuliaDiff/ChainRulesCore.jl/issues/349)